app/workers/zomato_data_generator.rb

```diff
@@ -0,0 +1,25 @@
+class ZomatoDataGenerator
+  def self.download_zomato_data(reader)
+    restaurants = reader.fetch_restaurants_list
+    restaurants.each do |restaurant|
```

Comment:
Note that exceptions may happen at any iteration, and it would be a shame to lose the entire fetching of 2000 restaurants just because you got an exception on restaurant #1999 .

You can use begin...rescue within the loop to catch exceptions for a single iteration.